src/CoreLib/DSeq.fs

```diff
@@ -89,23 +90,26 @@ and internal DStreamForwardDependencyType =
     /// Multicast a blob to all destinations in the Mapping
     | MulticastToNetwork of DependentDStream
 and internal DStreamFactory() = 
-    inherit HashCacheFactory<DStream>()
-    static let collectionByName = ConcurrentDictionary<_,_>(StringTComparer<int64>(StringComparer.Ordinal)) 
-    // Retrieve a DStream
-    static member ResolveDStream( hash ) = 
-        DStreamFactory.Resolve( hash ) 
+    static let collectionByJob = ConcurrentDictionary<Guid,(int64 ref)*ConcurrentDictionary<_,_>>() 
```

Comment:
/// A collection of DStream that serves as the network interface for intercommunication between Jobs. 
    /// The entry is indexed by jobID, while each entry holds a timestamp (in ticks of int64), of the recent input from that job. 
    /// That way, we can detect inactive jobs and cancel them later.